### DISCUSSION TITLE
[Feature] Mobile: filter by upload status

### DISCUSSION THREAD

**Discussion by Isti6github** [immich-app/immich#2155] [Feature Request]
### The feature

Filter option to show in the app the local, backuped(sync) and server pictures separetly.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by thumDer** [comment]
This feature would be rellay usefule, sometimes I want to revise my assets, if there is something I've missed, and seeing all assets just clutters the view, and very to easy to miss single local assets.

**Comment by snatch84** [comment]
+1 for this feature.
Sometimes I happen to delete photos directly from the Samsung gallery app..If it were possible to filter photos that are only on the server it would be easier to identify any useless photos for cleaning space

**Comment by xiongsp** [comment]
+1 For this. It's really helpful when revise my photos.
#16072

**Comment by dhex-** [comment]
Super necessary feature since erased files on server stays on mobile. Could use this along with date filter to find pictures erased on server to bulk delete them.

**Comment by la7eralus** [comment]
Would also love to see this feature. Especially since there is no two way sync available. So we could filter server only pictures and bulk delete them at once. These are the options, we need a filter for: https://immich.app/docs/FAQ/#what-is-the-difference-between-the-cloud-icons-on-the-mobile-app

**Comment by alexandrgorodilov** [comment]
I will also support this feature. I would really like to have such a filter to bring my gallery to perfection.

**Comment by mungo312** [comment]
+1, would be really great.

**Comment by bo0tzz** [comment]
Folks, please use the upvote button on the original post instead of "+1" comments.

**Comment by t00** [comment]
I barely started using immich and was looking for exactly such filter unsuccessfully. Seeing this issue means such feature is not yet implemented.

3 cloud icon + "all" buttons would be perfect to quickly tap on each to filter.

My use case is very simple, I just uploaded some camera shoots, there can be many days of photos. I did not upload all as some are not worth uploading or I need to leave them on device like ID photos and documents. Now I need to select all which were uploaded to delete from the device. This way I can miss some which were not uploaded by manually selecting them and then delete permanently.

Filter would be a life saver. Maybe even filter for photos which are/are not in any album, to pinpoint leftover photos?

One more usability issue is when uploading photos from device, there is no indication of the background upload action being carried out. What also happens is all selected albums are subsequently deselected after the upload succeeds, which forces the user to select a same bunch again to delete. Maybe there there is an opportunity to add "Upload & delete" or "Move to the cloud" feature which would make this process easier?

I can see filter is not much a priority as this issue is quite old, on the occasion of posting the above I am really curious how others manage uploads from the device followed by subsequent deletions from the device?

**Comment by orzechszek** [comment]
Would be a great feature. Especially I need a filter for not synced (only local?) photos, to verify which images/videos had a problem.

**Comment by MrAbelNovakerk** [comment]
Will this also give the possibility to filter the photo's created by phone an backed up to the server and the photos in external library

**Comment by EnderRobber101** [comment]
Also, the reverse: show photos that are not currently on the phone.

**Comment by Nopparuj-an** [comment]
Hi, I just started using Immich today and I have to say I like it so far.

My use case for Immich is to clear off my phone storage, offloading some to the server while still keeping most of the photos on the phone. Being able to browse my gallery on a computer and using deduplication tools, I removed a bunch of photos but then realized that they are still on the phone, and vice versa removing unneeded photos from the phone left a lot of garbage on the server.

Having ability to filter by “only on this device” and “only on the server” solves this problem directly and efficiently.

**Comment by tablitam** [comment]
This feature request seems to be coming over and over again since long time ago and from many users. The status for each image is already there (cloud with tick -both in server and device-, empty cloud -only on server-, crossed cloud -only on device-), it seems that the only thing needed is to filter by these existing statuses on display. Is there a reason this hasn't been included in the mobile app yet?

**Comment by sehmee** [comment]
I was looking for a feature like this as well, unfortunately all existing requests point to this one from 2023. I upvoted with the hope it will be added soon.

**Comment by j05hp** [comment]
Just want to add. I have had Immich possibly almost a year now and I have been waiting for this feature the entire time. There are now thousands of junk photos that I have deleted from my iPhone but are stuck on Immich. Hope this feature comes soon.

I use Immich solely with a phone and access via my PC. 

Another feature that would tie in great would be a state in Immich to tell you where the photos have been synced from. Then a filter to see that they don't exist on that device and are now cloud only images.

**Comment by sebnil92** [comment]
Yes, same here. My workflow is to review photos via my Mac and remove videos and photos from there. If I only could filter those cloud only photos the sync to Immich would be super quick.

**Comment by omar-salama** [comment]
i actually need this feature, and since it’s been raised since forever without any action, i will make it myself. someone poke me in a couple of days as of rn i am busy.

**Reply by iswaterbad** [reply]
i just found this thread, and wanted to leave a comment to remind you since it's been a few days since you wrote this. this would be an amazing feature to have!
